storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/utils/LogviewerResponseBuilder.java

```diff
@@ -79,12 +79,20 @@ public static Response buildSuccessJsonResponse(Object entity, String callback,
      */
     public static Response buildDownloadFile(File file, Meter numFileDownloadExceptions) throws IOException {
         try {
+            String fname;
+            try {
+                String topoInfo = file.getParentFile().getParentFile().getName();
+                String port = file.getParentFile().getName();
+                fname = String.format("%s-%s-%s", topoInfo, port, file.getName());
```

Comment:
we should probably have supervisor/hostname added to this.

Reply:
Added